cheroot/wsgi.py

```diff
@@ -137,6 +137,9 @@ def respond(self):
                     raise ValueError('WSGI Applications must yield bytes')
                 self.write(chunk)
         finally:
+            # Send headers if not already sent
+            if not self.req.sent_headers:
```

Comment:
It's always paired with "if not sent" check. Try moving it inside the method and maybe rename it into smth like `ensure_headers_sent` or so